firebase-common/src/main/java/com/google/firebase/Timestamp.kt

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase
+
+import android.os.Build
+import android.os.Parcel
+import android.os.Parcelable
+import androidx.annotation.RequiresApi
+import java.time.Instant
+import java.util.Date
+
+/**
+ * A Timestamp represents a point in time independent of any time zone or calendar.
+ *
+ * Represented as seconds and fractions of seconds at nanosecond resolution in UTC Epoch time. It is
+ * encoded using the Proleptic Gregorian Calendar which extends the Gregorian calendar backwards to
+ * year one. Furthermore,It is encoded assuming all minutes are 60 seconds long, specifically leap
+ * seconds are "smeared" so that no leap second table is needed for interpretation. Range is from
+ * 0001-01-01T00:00:00Z to 9999-12-31T23:59:59.999999999Z. By restricting to that range, we ensure
+ * that we can convert to and from RFC 3339 date strings.
+ *
+ * @see <a href="https://git.page.link/timestamp-proto">Timestamp</a>The ref timestamp definition
```

Comment:
```suggestion
 * @see <a href="https://git.page.link/timestamp-proto">Timestamp</a> The ref timestamp definition
```

Reply:
This is actually done on purpose. Devsite separates the link from the content- so adding a space makes it look weird.

[See the Timestamp page on devsite for reference.](https://firebase.google.com/docs/reference/android/com/google/firebase/Timestamp)